lib/regexp_parser/scanner/scanner.rl

```diff
@@ -120,7 +120,7 @@
 
   literal_delimiters    = ']' | '}';
 
-  ascii_print           = ((0x20..0x7e) - meta_char);
+  ascii_print           = ((0x20..0x7e) - meta_char - '#');
```

Comment:
it took me a bit to figure out why this is necessary.

it is because the `(ascii_print -- space)+` [...] literal matcher used to match up to and including pound signs, and the `comment` matcher only had a chance to fire if the literal matcher was interrupted by a space.

having the declarations of what can "safely" be considered a literal spread out is a little confusing, but that was already the case before this PR. i might look into consolidating this a bit later.